packages/template-retail-react-app/app/pages/account/wishlist/partials/wishlist-primary-action.test.js

```diff
@@ -32,19 +36,56 @@ beforeEach(() => {
     jest.resetModules()
 })
 
-test('renders primary action component', async () => {
-    const {getByRole} = renderWithProviders(
-        <MockedComponent variant={mockedCustomerProductListsDetails.data[0]} />
-    )
+test('the Add To Cart button', async () => {
+    const variant = mockedCustomerProductListsDetails2.data[3]
+    const {getByRole} = renderWithProviders(<MockedComponent variant={variant} />)
+
     const addToCartButton = getByRole('button', {
         name: /add to cart/i
     })
     expect(addToCartButton).toBeInTheDocument()
-    user.click(addToCartButton)
+    userEvent.click(addToCartButton)
 
     await waitFor(() => {
         // Chakra UI renders multiple elements with toast title in DOM for accessibility.
         // We need to assert the actual text within the alert
-        expect(screen.getByRole('alert')).toHaveTextContent(/items added to cart/i)
+        expect(screen.getAllByRole('alert')[0]).toHaveTextContent(/1 item added to cart/i)
+    })
+})
+
+test('the Add Set To Cart button', async () => {
+    const productSetWithoutVariants = mockedCustomerProductListsDetails2.data[1]
+    renderWithProviders(<MockedComponent variant={productSetWithoutVariants} />)
+
+    const button = screen.getByRole('button', {name: /add set to cart/i})
+    userEvent.click(button)
+
+    await waitFor(() => {
+        expect(screen.getAllByRole('alert')[0]).toHaveTextContent(/2 items added to cart/i)
     })
 })
+
+test('the View Full Details button', () => {
```

Comment:
I realize it's unlikely to matter for such a tiny component, but I might still use `async` here and `waitFor` just to make sure the tests don't behave differently or fail in CI / different OS runtimes